lib/phoenix_live_view/engine.ex

```diff
@@ -427,20 +436,29 @@ defmodule Phoenix.LiveView.Engine do
   ## Optimize possible expressions into live structs (rendered / comprehensions)
 
   defp to_live_struct({:for, _, [_ | _]} = expr, vars, _assigns, caller) do
-    with {:for, meta, [_ | _] = args} <- expr,
+    with {:for, meta, [gen | args]} <- expr,
          {filters, [[do: {:__block__, _, block}]]} <- Enum.split(args, -1),
          {dynamic, [{:safe, static}]} <- Enum.split(block, -1) do
       {block, static, dynamic, fingerprint} =
         analyze_static_and_dynamic(static, dynamic, taint_vars(vars), %{}, caller)
 
-      for = {:for, meta, filters ++ [[do: {:__block__, [], block ++ [dynamic]}]]}
+      gen_var = Macro.var(:for, __MODULE__)
```

Comment:
```suggestion
      gen_var = Macro.unique_var(:for, __MODULE__)
```

Will reduce the chance of conflicts in nested `for`.